week-1/Homework/extra/2-piping.js

```diff
@@ -16,43 +16,51 @@
   the final result to the variable goodCode
 */
 
-function add() {
-
+function add(a, b) {
+  const total = a + b;
+  return Number(total.toFixed(2));
 }
 
-function multiply() {
-
+function multiply(a, b) {
+  const total = a * b;
+  return Number(total.toFixed(2));
```

Comment:
Notice that, by making `add()` and `multiply()` return a result truncated to two decimals, we unnecessarily lose precision when we combine them (like we do below). It'd be a better practice to let the formatting aspects to `format()` :)

Reply:
Noted, I totally agree! But again the case 2 test for (Add) contains a decimal number, so if I remove it from there the test will fail.. Unless there is another way/method that I can use instead of toFixed()?

Reply:
You're right (again).

IMO the test is implemented too naively, float numbers (which is how JS stores numbers with decimals) are not precise and you can't simply compare them to other floats with `===`.

Instead, you want to check if the two floats are close enough. The test could've been implemented like this:

```js
// `tolerance` is the maximum difference we allow between the two numbers. Notice
// we have to provide it on every comparison as there is no one size fits all.
const floatsEqual = (a, b, tolerance) => Math.abs(a - b) < tolerance;

// Here for example we decide that these two floats are equal if they are no more 
// than 0.1 different
test("add function - case 2 works", floatsEqual(add(2.4, 5.3), 7.7, 0.1));
```

In reality there are not many cases where you actually need to treat floats as precise numbers. If you find yourself needing that, you may want consider avoiding floats altogether (for example, when working with money is common to represent amounts in cents so that you don't have decimals).